### PR TITLE
fix toc resolve bug: when toc link has anchor, tocrel might not resolve correctly

### PR DESCRIPTION
**Pull request by ansyral**

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.TableOfContents/TocDocumentProcessor.cs [hunk 1]
```diff
@@ -159,14 +159,14 @@ private void RegisterTocMap(TocItemViewModel item, string key, IDocumentBuildCon
             var tocHrefType = Utility.GetHrefType(tocHref);
             if (tocHrefType == HrefType.MarkdownTocFile || tocHrefType == HrefType.YamlTocFile)
             {
-                context.RegisterToc(key, tocHref);
+                context.RegisterToc(key, UriUtility.GetPath(tocHref));
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.TableOfContents/TocDocumentProcessor.cs]: should we give a warning for reference a toc with anchor?

**Comment by superyyrrzz** [comment]
:shipit:

**Comment by vicancy** [comment]
Thanks 😸

**Comment by hellosnow** [comment]
:shipit:

**Pull request merged**